File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/template/TemplatePattern.java
package jv.com.tencent.cain.design_pattern.template;

/**
 * @author : jiangyu
 * @date : 2019/11/3
 * @desc : 模版设计模式
 */
public class TemplatePattern {
    public static void main(String[] args) {
        TemplateAbstract templateAbstract;
        templateAbstract = new TemplateConcrete1();
        templateAbstract.templateMethod();
        templateAbstract = new TemplateConcrete2();
        templateAbstract.templateMethod();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/template/TemplatePattern.kt
package kt.com.tencent.cain.desigin_pattern.template

/**
 * @author : jiangyu
 * @date   : 2019/11/3
 * @desc   : 模版设计模式
 */
fun main() {
    lateinit var template: TemplateAbstract
    template = TemplateConcrete1()
    template.templateMethod()
    template = TemplateConcrete2()
    template.templateMethod()
}